app/controllers/api/v1/report/general_controller.rb

```diff
@@ -0,0 +1,18 @@
+class Api::V1::Report::GeneralController < Doorkeeper::ApplicationController
+  before_action -> { doorkeeper_authorize! :reporting_access }
+
+  respond_to :json
+
+  def show
+    start_date = params.fetch(:start_date, Time.zone.parse("15:00:00") - 1.day)
```

Comment:
Is `1.day.ago( Time.zone.parse("15:00:00") )` any easier to ready?

Reply:
I'll swap those two lines and make it `1.day.before(end_date)`